src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -162,37 +162,56 @@ class GoogleExtensions(
   override def publishGroup(id: WorkbenchGroupName): Future[Unit] =
     googlePubSubDAO.publishMessages(googleServicesConfig.groupSyncTopic, Seq(id.toJson.compactPrint))
 
-  override def onGroupUpdate(groupIdentities: Seq[WorkbenchGroupIdentity]): Future[Unit] =
-    onGroupUpdateRecursive(groupIdentities, Seq.empty).unsafeToFuture()
-
-  private def onGroupUpdateRecursive(groupIdentities: Seq[WorkbenchGroupIdentity], visitedGroups: Seq[WorkbenchGroupIdentity]): IO[Unit] =
+  override def onGroupUpdate(groupIdentities: Seq[WorkbenchGroupIdentity]): Future[Unit] = {
```

Comment:
Could add a comment to document the how/why we update what we do when we update a group?  You and Doug spent a fair amount of time talking about why we do the group sync and how/why we need to grab ancestors of Group Y but not of Group X, etc.  Furthermore, all of this update stuff, IIRC, is because Google needs to store the intersection groups.  Sam doesn't care about the intersection groups (I think?) because that's basically what LDAP does for us, is calculate intersections of groups.  You and Doug understand how this logic works right now, and understand WHY we need to do it this way.  In 6 months, if someone other than you or Doug needs to modify this, will they understand why we're doing it this way so as not to break it?

Reply:
good point!

Reply:
Awesome!